### DISCUSSION TITLE
pause/resume reviewing/scan & load cache.db

### DISCUSSION THREAD

**Discussion by hejhopsa** [theophanemayaud/video-simili-duplicate-cleaner#80] [Q&A]
Hello, 
First of all thank you for the software 
I have a huge loads of videos with lots of duplicates. The first scan took ages to make. I've verified some of videos and the app crashed or I've closed it (Im not really sure). I have "cache.db" file which is 500 mb. Can I somehow load it again to review the videos that were indexed?

Im willing to donate! :-)

**Comment by theophanemayaud** [comment]
Hi ! Thanks, it’s really my pleasure ☺️

In any case, it should use the existing cache file. Did you leave it to a default location or manually select it ? If it’s the default, it should reuse it automagically. And if you manually set the location of the cache file, you can simply select it again. On windows it might warn you that it will overwrite, but that shouldn’t actually be the case. 

Keep in mind however (and that is a good feature suggestion!) there is no way to « use only cached videos ». That is, even if videos are cached the app will still check if new videos exist (and new videos might actually mean videos it didn’t succeed to cache, which it will check again and that might take time…). 

So, you should be able to use the cache again, it will greatly improve time (how many videos do you have for such a large cache ? 🤓). However it might still take some time to check new videos.

**Reply by hejhopsa** [reply]
> Hi ! Thanks, it’s really my pleasure ☺️
<3
> 
> In any case, it should use the existing cache file. Did you leave it to a default location or manually select it ? If it’s the default, it should reuse it automagically. And if you manually set the location of the cache file, you can simply select it again. On windows it might warn you that it will overwrite, but that shouldn’t actually be the case.
> 

> > Keep in mind however (and that is a good feature suggestion!) there is no way to « use only cached videos ». That is, even if videos are cached the app will still check if new videos exist (and new videos might actually mean videos it didn’t succeed to cache, which it will check again and that might take time…).

**So if I understand correctly - I need to rescan anyway? The way I've imagined it was that I can somehow resume reviewing what has been already added to db**

> So, you should be able to use the cache again, it will greatly improve time (how many videos do you have for such a large cache ? 🤓). However it might still take some time to check new videos.

 ... around 25 000 files. 1 TB🤓

Thank you for the answer! So I've installed it previously in different directory. I believe it was also the app from the original creator. It was a year ago or so but I wanted to get back to it and resume cleaning. Today I bought the app from your store and installed it. But I needed to transfer it through accounts because I have everything blocked including Microsoft Store on the main one (I write this just in case this might be the issue). 

**What is the default directory then?** 

Anyway I keep getting "error" message 
![image](https://user-images.githubusercontent.com/41187733/192094868-f7578870-e934-4840-872d-1ae4f7f5d5a4.png)
Maybe because Windows Apps folder is restricted C:\Program Files\WindowsApps\ (I couldn't choose different one when installing from MS Store)
Even when I have successful scan and I close the app and reopen it again this window pops up again and I need to OVERWRITE the previous DB file. Progress seems to be lost (or I need to rescan to see the review window again?). 
**"The file cache.db already exists. Do you want to overwrite it?"**

**Reply by theophanemayaud** [reply]
> So if I understand correctly - I need to rescan anyway? The way I've imagined it was that I can somehow resume reviewing what has been already added to db

The cache is for video thumbnails and metadata. When scanning again, it will check what videos are on disk, and if it sees a match in the cache it will skip thumbnails and metadata retrieval which speeds up considerably the scan. But it still scans.
I'm writing down the idea you're implying, of only using cached data. That would indeed be great in the case of very very large number of videos. There are a few hurdles because of the current implementation. If only cached data is used, we shouldn't be able to change the thumbnails type but only keep the cached ones, but otherwise it can be simply done with a "Only use cached data" switch, which would skip the scan phase and directly go to the comparison phase. Thanks for the great idea !! It'll also help me in my workflows a lot ! I don't have 1TB but still > 200GB which does take ~10 mins to rescan even when cached. Without cache it's more like 30 mins to an hour, but your method could even reduce further.

> Anyway I keep getting "error" message

Oh yes that's a bug on windows I haven't managed to consistently solve apparently... I'll reopen #69 !

> "The file cache.db already exists. Do you want to overwrite it?"
 Write

That's also a bug but only of the prompt. It doesn't in fact overwrite the cache file, instead it does use it again. You can check by looking at the cache file size. After the "Do you want to overwrite it?" -> yes confirmation, it shouldn't have changed size. If it's ~500MB after a session, and when you re-select it on another session, after you "confirm overwrite it" it still should be ~500MB. But as explained above, the cache isn't exclusively used, which might explain why it's not clear to your that there is speed improvement because of the existing cache. I just created #82 to remember to fix this false warning.

I hope my explanations are clear :)

**Comment by theophanemayaud** [comment]
Also by the way, if you figure out wether it’s really crashing and find out what could be causing the crash, please send me more details so I can investigate and fix it ! I don’t have a log of errors or crash details for now though so it’s just details you figure out that could help 😬

**Reply by hejhopsa** [reply]
Im not sure about the crash but if it happens I'll send you the log. The only reason I can imagine is db file too big.
